### PR TITLE
Username as email fix

### PR DESCRIPTION
**Pull request by jasonify**

### CODE REVIEW

File: server/controllers/user.controller.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ function update(req, res, next) {
       user.avatarUrl = `https://${S3_BUCKET}.s3.amazonaws.com/${user._id}`
     }
     return user.save().then((newUser) => {
-      const userMinusPassword = Object.assign(newUser, {password: null});
+      const userMinusPassword = Object.assign({}, newUser, {password: null});
```
Comment (crablar) [inline server/controllers/user.controller.js]: Was this the bug that resulted in null passwords?

Comment (jasonify) [reply server/controllers/user.controller.js]: This had been the area yeah but it looked like this:

user.save() // This is async so next line will execute before this completes, causing the bug
user.password = null
res.json(user);

Comment (crablar) [reply server/controllers/user.controller.js]: Wow, async bug, that is nasty.

Comment (TheHollidayInn) [reply server/controllers/user.controller.js]: Yea, all mongo operations are promises except queries need to use exec.

**Pull request merged**